bindata/v4.1.0/alerts/kube-apiserver-slos.yaml

```diff
@@ -0,0 +1,429 @@
+          label_replace(
+            sum(rate(apiserver_request_duration_seconds_count{job="apiserver",verb=~"LIST|GET",subresource!~"proxy|log|exec",scope=~"resource|"}[5m]))
+            -
+            (
+              sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",verb=~"LIST|GET",subresource!~"proxy|log|exec",scope=~"resource|",le="0.1"}[5m]))
+              or
+              vector(0)
+            )
+          , "type", "slow-resource", "_none_", "")
+          or
+          # namespace-scoped latency
+          label_replace(
+            sum(rate(apiserver_request_duration_seconds_count{job="apiserver",verb=~"LIST|GET",scope=~"namespace"}[5m]))
+          - sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",verb=~"LIST|GET",scope="namespace",le="0.5"}[5m]))
+          , "type", "slow-namespace", "_none_", "")
+          or
+          # cluster-scoped latency
+          label_replace(
+            sum(rate(apiserver_request_duration_seconds_count{job="apiserver",verb=~"LIST|GET",scope="cluster"}[5m]))
+            - sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",verb=~"LIST|GET",scope="cluster",le="5"}[5m]))
+          , "type", "slow-cluster", "_none_", "")
+        ) / scalar(sum(rate(apiserver_request_total{job="apiserver",verb=~"LIST|GET"}[5m])))
+      labels:
+        verb: read
+      record: apiserver_request:burnrate5m
+    - expr: |
+        (
+          # error
+          label_replace(sum(rate(apiserver_request_total{job="apiserver",verb=~"LIST|GET",code=~"5.."}[30m])), "type", "error", "_none_", "")
+          or
+          # resource-scoped latency
+          label_replace(
+            sum(rate(apiserver_request_duration_seconds_count{job="apiserver",verb=~"LIST|GET",subresource!~"proxy|log|exec",scope=~"resource|"}[30m]))
+            -
+            (
+              sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",verb=~"LIST|GET",subresource!~"proxy|log|exec",scope=~"resource|",le="0.1"}[30m]))
+              or
+              vector(0)
+            )
+          , "type", "slow-resource", "_none_", "")
+          or
+          # namespace-scoped latency
+          label_replace(
+            sum(rate(apiserver_request_duration_seconds_count{job="apiserver",verb=~"LIST|GET",scope=~"namespace"}[30m]))
+          - sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",verb=~"LIST|GET",scope="namespace",le="0.5"}[30m]))
+          , "type", "slow-namespace", "_none_", "")
+          or
+          # cluster-scoped latency
+          label_replace(
+            sum(rate(apiserver_request_duration_seconds_count{job="apiserver",verb=~"LIST|GET",scope="cluster"}[30m]))
+            - sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",verb=~"LIST|GET",scope="cluster",le="5"}[30m]))
+          , "type", "slow-cluster", "_none_", "")
+        ) / scalar(sum(rate(apiserver_request_total{job="apiserver",verb=~"LIST|GET"}[30m])))
```

Comment:
Actually, with the resource exclusion, this denominator needs to move to each OR, and then on resource scoped you need to exclude subresource from the denominator `(a or b or c)/d` -> `a/d or b/d or c/d` -> `a (exclude resource) / d (exclude resource) or b/d or c/d`

Reply:
should be fixed now. i added the denominator to the error contribution too.